probe-rs/src/architecture/arm/mod.rs

```diff
@@ -56,40 +51,30 @@ pub enum ArmError {
     },
 
     /// An error occurred while using a debug port.
-    #[error("Error using a debug port.")]
     DebugPort(#[from] DebugPortError),
 
     /// The core has to be halted for the operation, but was not.
-    #[error("The core needs to be halted for this operation but was not")]
     CoreNotHalted,
 
     /// Performing certain operations (e.g device unlock or Chip-Erase) can leave the device in a
     /// state that requires a probe re-attach to resolve.
-    #[error("Probe and device internal state mismatch. A probe re-attach is required.")]
     ReAttachRequired,
 
-    /// An operation was not performed because the required permissions were not given.
+    /// An operation could not be performed because it lacked the permission to do so: {0}.
```

Comment:
This might be a bit verbose to display as an error. You may add `#[ignore_extra_doc_attributes]` so that only the first line will be part of the error message.

Reply:
I thought it’d do that by default.

Reply:
displaydoc would IIRC simply fail in this situation, docsplay can handle all the docs you give it. Since the library is not only meant for error messages, it defaults to displaying everything.

Reply:
BTW, I never used it before, I just saw it used in another module and figured this was a good way to reduce the amount of noise in the code.

Reply:
Yeah, it's my fork of displaydoc because I had a few issues with it, but it's not perfect. The formatting placeholders can easily introduce inelegant noise into the documentation, and automatically trying to fix that is pretty hard.